Hometask 5

kkonanov opened the pull request.

Commit:
[task2]
add interface Actionable
add functional interface Action
release inner iterator by method execute(Action action)
add unit-tests for DoorEventHandler, LightEventHandler, AllLightEventHandler, SmartHome.execute, Room.execute, Light.execute, Door.execute

Commit:
[task3]
add Signalization and State

Commit:
[task3]
add SignalizationEventHandler and Signalization

kkonanov committed:
[task3]

kkonanov committed:
[task4]
add Spring realization and using API

kkonanov committed:
[task4]

kkonanov committed:
[task5]

kkonanov committed:
delete @Bean eventHandlers
create ConversionFactory
change SensorEventsManagerAdapter

a-ivanov commented on src/main/java/ru/sbt/mipt/oop/RemoteControlImp.java:
```diff
@@ -0,0 +1,37 @@
+		return buttons;
+	}
+
+	public void setButton(String buttonCode, Command command) {
```
Зачем сеттер? Лишний раз их лучше не использовать.
Можно просто передавать готовую мару аргументом сюда.

a-ivanov commented on src/main/java/ru/sbt/mipt/oop/RemoteControlImp.java:
```diff
@@ -0,0 +1,37 @@
+		this.rcId = rcId;
+	}
+
+	public Map<String, Command> getButtons() {
```
Зачем выдавать наружу детали реализации класса?

a-ivanov commented on src/main/java/ru/sbt/mipt/oop/RemoteControlImp.java:
```diff
@@ -0,0 +1,37 @@
+import java.util.Objects;
+import java.util.Set;
+
+public class RemoteControlImp implements RemoteControl {
```
Этот класс должен быть добавлен в спринг конфигурацию приложения, см. https://github.com/iagoshkov/smart-home-2020/blob/master/STUDENT_GUIDE.md#задание-5

a-ivanov commented on src/main/java/ru/sbt/mipt/oop/Application.java:
```diff
@@ -1,27 +1,16 @@
-        application.run();
+        AbstractApplicationContext context = new AnnotationConfigApplicationContext(SmartHomeConfiguration.class);
+        SensorEventsManager sensorEventsManager = context.getBean(SensorEventsManager.class);
+        Command command = (Command) context.getBean("closeDoorHallCommand");
```
Зачем? Команды должен вызывать RemoteControl, который, в свою очередь, вызывает RemoteControlRegistry.
Уберите эту строчку.

kkonanov committed:
[hometask-4]
delete SensorEventsManagerAdapter.java
add EventHandlerAdapter (implements com.coolcompany.smarthome.events.EventHandler)
change configuration file

kkonanov committed:
[hometask-5]
add RemoteControlConfiguration

kkonanov committed:
Merge remote-tracking branch 'origin/hometask-5' into hometask-5

# Conflicts:
#	src/main/java/ru/sbt/mipt/oop/Application.java
#	src/main/java/ru/sbt/mipt/oop/RemoteControlImp.java
#	src/main/java/ru/sbt/mipt/oop/SmartHomeConfiguration.java
#	src/main/java/ru/sbt/mipt/oop/command/ActivateSignalizationCommand.java
#	src/main/java/ru/sbt/mipt/oop/command/AlarmModeSignalizationCommand.java
#	src/main/java/ru/sbt/mipt/oop/command/CloseDoorHallCommand.java
#	src/main/java/ru/sbt/mipt/oop/command/LightOffAllHomeCommand.java
#	src/main/java/ru/sbt/mipt/oop/command/LightOnAllHomeCommand.java
#	src/main/java/ru/sbt/mipt/oop/command/LightOnHallCommand.java
#	src/test/java/ru/sbt/mipt/oop/RemoteControlImpTest.java
#	src/test/java/ru/sbt/mipt/oop/command/ActivateSignalizationCommandTest.java
#	src/test/java/ru/sbt/mipt/oop/command/AlarmModeSignalizationCommandTest.java

a-ivanov commented on src/main/java/ru/sbt/mipt/oop/RemoteControlConfiguration.java:
```diff
@@ -0,0 +1,68 @@
+
+@Configuration
+@Import(SmartHomeConfiguration.class)
+public class RemoteControlConfiguration {
```
Эту конфигурацию необходимо в контексте в main добавить.

kakonanov commented:
Ок, спасибо
